components/AttackSlot.jsx

```diff
@@ -0,0 +1,69 @@
+import styled from "styled-components";
+
+export const AttackSlot = ({ register, index, attackRemove }) => {
```

Comment:
I am still thinking about these function names (attackRemove, etc.).
Following naming conversion the action that is performed would generally come first. This would mean:

- `removeAttack()`
- `insertAttack()`
- `attackFields` (not an action)

This reads more naturally but it makes things that belong together (e.g. all attack related functions) start with different words.

Therefore I guess the solution I'd use would be to keep the Object returned by `useFieldArray` and just pass that around.

- `attack.remove()`
- `attack.insert()`
- `attack.fields`